helsingborg-stad/helsingborg-io-sls-api
# feat(pdf-ms): add a pdf generator ms

# PDF generator microservice

This microservice generates pdfs from case data, and saves it (as a buffer) on the case. 
Check the readme for some explanation of the data flow and template system. 

## How to deploy
Be sure to first deploy the pdfTemplates resource in the helsingborg-io-sls-resources repo, so that the required S3 bucket exists.
Then to deploy it, just run yarn in the service folder, and then deploy with serverless as normal, sls deploy. 

In the folder templateExample there is two template files that exemplifies how the template system works. These should be put into the S3 bucket named "pdf-case-templates-s3-{some-id}" for the generator to work. 

## How to test
To trigger the service, you need to update a case, setting its status to 'submitted'. So you have to do this, either through the app, or through a direct put-request against the case api. To view the generated pdf, you can go to the S3 bucket "pdf-case-templates-..." and it should be in the /cases/-folder. It's also saved on the case in the db, but as a buffer so you have to do some conversion to be able to view it from there. 

Then beware that it will only generate a pdf for a case once; unless you go into the db and switch the pdfGenerated value to false.
